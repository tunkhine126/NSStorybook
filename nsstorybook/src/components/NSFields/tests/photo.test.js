/**
 *
 * Tests for Photo
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import TestingComponent from 'components/shared/Testing';
import Photo from '../photo';

describe('<Photo />', () => {
  it('Expect no errors rending Photo component', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <Photo field="photo" label="Photo Test" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <Photo field="photo" label="Photo Test" />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
