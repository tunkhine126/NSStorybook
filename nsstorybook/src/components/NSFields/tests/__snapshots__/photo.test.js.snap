// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Photo /> Should render and match the snapshot 1`] = `
<div
  class="makeStyles-dropzone-156 dropzone"
  tabindex="0"
>
  <input
    accept="image/jpeg, image/png, image/tiff, image/ief, image/gif"
    autocomplete="off"
    multiple=""
    style="display: none;"
    tabindex="-1"
    type="file"
  />
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-icon-157"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <circle
        cx="12"
        cy="12"
        r="3.2"
      />
      <path
        d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
      />
    </svg>
    <p
      class="MuiTypography-root makeStyles-caption-155 MuiTypography-body1 MuiTypography-gutterBottom"
    >
      Photo Test
    </p>
  </div>
</div>
`;
